src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -479,10 +526,12 @@ private PSObject[] InvokePowerShell(string command, IDictionary<string, object>
                     powerShell.AddParameter(kvp.Key, kvp.Value);
                 }
 
-                var result = new PSObject[0];
+                PowerShellResult result = null;
                 try
                 {
-                    result = powerShell.Invoke()?.ToArray();
+                    PSObject[] output = powerShell.Invoke()?.ToArray();
+                    ErrorRecord[] errors = powerShell.Streams?.Error?.ToArray();
+                    result = new PowerShellResult(output, errors, powerShell.HadErrors);
```

Comment:
I notice that both output and errors could potentially be set to null here... Is that going to be ok?

Reply:
Looks like we think they can't actually be `null`. Would be so much nicer if the type system were able to track that...